src/test/java/LocationsUtilityTest.java

```diff
@@ -0,0 +1,121 @@
+  private static final List<Task> TASKS_WITH_TWO_LOCATIONS =
+      ImmutableList.of(TASK_WITH_TWO_LOCATIONS);
+  private static final List<Task> TASKS_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY =
+      ImmutableList.of(TASK_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY);
+  private static final List<Task> TASKS_WITH_EMPTY_LOCATION =
+      ImmutableList.of(TASK_WITH_EMPTY_LOCATION);
+  private static final List<Task> ALL_TASKS =
+      ImmutableList.of(
+          TASK_WITH_NO_NOTES,
+          TASK_WITH_NO_LOCATION,
+          TASK_WITH_ONE_LOCATION,
+          TASK_WITH_TWO_LOCATIONS,
+          TASK_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY,
+          TASK_WITH_EMPTY_LOCATION);
+
+  @Test
+  public void getLocationNoTasks() {
+    // Obtain locations in the the task notes of no tasks.
+    Assert.assertEquals(ImmutableList.of(), LocationsUtility.getLocations(PREFIX, NO_TASKS));
+  }
+
+  @Test
+  public void getLocationNoNotes() {
+    // Obtain location in the task notes of one task with no notes defined.
+    Assert.assertEquals(
+        ImmutableList.of(), LocationsUtility.getLocations(PREFIX, TASKS_WITH_NO_NOTES));
+  }
+
+  @Test
+  public void getNoLocation() {
+    // Obtain location in the task notes of one task with no location.
+    Assert.assertEquals(
+        ImmutableList.of(), LocationsUtility.getLocations(PREFIX, TASKS_WITH_NO_LOCATION));
+  }
+
+  @Test
+  public void getOneLocation() {
+    // Obtain location in the task notes of one task with one location.
+    Assert.assertEquals(
+        ImmutableList.of("Google Kitchener"),
+        LocationsUtility.getLocations(PREFIX, TASKS_WITH_ONE_LOCATION));
+  }
+
+  @Test
+  public void getTwoLocations() {
+    // Obtain location in the task notes of one task with two locations. Second location, Google
+    // Montreal, is ignored.
+    Assert.assertEquals(
+        ImmutableList.of("Google Kitchener"),
+        LocationsUtility.getLocations(PREFIX, TASKS_WITH_TWO_LOCATIONS));
+  }
+
+  @Test
+  public void getIncorrectlyEnclosedLocation() {
+    // Obtain location in the task notes of one task with one location enclosed with () instead of
+    // [].
+    Assert.assertEquals(
+        ImmutableList.of(),
+        LocationsUtility.getLocations(PREFIX, TASKS_WITH_ONE_LOCATION_ENCLOSED_INCORRECTLY));
+  }
+
+  @Test
+  public void getEmptyLocation() {
+    // Obtain location in the task notes of one task with one location with [Location: ] tag but
+    // nothing inside of it.
+    Assert.assertEquals(
+        ImmutableList.of(""), LocationsUtility.getLocations(PREFIX, TASKS_WITH_EMPTY_LOCATION));
+  }
+
+  @Test
+  public void getLocationAllTasks() {
+    // Obtain location in the task notes of five tasks each with either one, empty or no location as
+    // noted in the individual tests above.
+    Assert.assertEquals(
+        ImmutableList.of("Google Kitchener", "Google Kitchener", ""),
+        LocationsUtility.getLocations(PREFIX, ALL_TASKS));
```

Comment:
Would it make sense to remove duplicate locations in the output? I have no strong opinion either way - it just seems like you are going to have to do this deduplication anyways.

Reply:
Yea I can see why it makes sense to deduplicate in the `getLocations` method itself plus it seems like adding `.distinct()` to `.stream()` did the deduplication perfectly. Thanks for the suggestion!